python_modules/libraries/dagster-aws/dagster_aws/s3/sensor.py

```diff
@@ -1,43 +1,100 @@
-from typing import Optional
+from datetime import datetime
+from typing import Any, List, Optional, cast
 
 import boto3
 import dagster._check as check
+from dagster._annotations import deprecated
+
+from dagster_aws._stubs import ObjectTypeDef
 
 
 class ClientException(Exception):
     pass
 
 
-def get_s3_keys(
+def get_objects(
     bucket: str,
     prefix: str = "",
     since_key: Optional[str] = None,
-    s3_session: Optional[boto3.Session] = None,
-):
+    since_last_modified: Optional[str] = None,
+    client=None,
+) -> List[ObjectTypeDef]:
+    """Retrieves a list of object keys in S3 for a given `bucket`, `prefix`, and filter option.
+
+    Args:
+        bucket (str): s3 bucket
+        prefix (str): s3 object prefix
+        since_key (Optional[str]): retrieve objects after the modified date of this key
+        since_last_modified (Optional[str]): retrieve objects after this timestamp
```

Comment:
Looks like this is ISO format, but it could be helpful to include that in the docstring.